libp2p/6-interfaces.md

```diff
@@ -1,139 +1,99 @@
 6 Interfaces
 ============
 
+`libp2p` is a conglomerate of several protocols working together to offer a common solid interface with talking with any other network addressable process. This is made possible by shimming current exist protocols and implementations through a set of explicit interfaces, from Peer Routing, Discovery, Stream Muxing, Transports, Connections and so on.
+
 ## 6.1 libp2p
 
+libp2p, the top module that interfaces all the other modules that make a libp2p instance, must offer an interface for dialing to a peer and plugging in all of the modules (e.g. which transports) we want to support. We present libp2p interface and UX on setion 6.6, after presenting every other module interface.
+
 ## 6.2 Peer Routing
 
-## 6.3 Swarm
+![](https://raw.githubusercontent.com/diasdavid/abstract-peer-routing/master/img/badge.png)
 
-~~The network is abstracted through the swarm which presents a simplified interface for the remaining layers to have access to the network. This interface should look like:~~
+A Peer Routing service offers a way for libp2p Node to find the PeerInfo of another Node, so that it can dial to that node. In it is most pure form, a Peer Routing module should have a interface that given a 'key', a set of PeerInfos are returned.
+See https://github.com/diasdavid/abstract-peer-routing for the interface and tests.
```

Comment:
this LGTM. 

we may have to rethink some of it when we address peer packet switching.